### ISSUE TITLE
error when compiling

### ISSUE THREAD

**Issue by adriantdlg** [HexHive/FuZZan#3]
Hi all,

I get this error when compiling:

1 warning generated.
[1317/3692] Building CXX object projects/compiler-rt/lib/sanitizer_common/CMakeFiles/RTSanitizerCommonNoTermination.x86_64.dir/sanitizer_platform_limits_posix.cc.o
FAILED: projects/compiler-rt/lib/sanitizer_common/CMakeFiles/RTSanitizerCommonNoTermination.x86_64.dir/sanitizer_platform_limits_posix.cc.o 
/usr/local/bin/clang++  -DHAVE_RPC_XDR_H=0 -DHAVE_TIRPC_RPC_XDR_H=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Iprojects/compiler-rt/lib/sanitizer_common -I/home
/adrian/Downloads/fuzzers/FuZZan/LLVM/llvm/projects/compiler-rt/lib/sanitizer_common -I/usr/include/libxml2 -Iinclude -I/home/adrian/Downloads/fuzzers/FuZZan/LLVM/llvm/include -I/home/adrian/Downloads/fu
zzers/FuZZan/LLVM/llvm/projects/compiler-rt/lib/sanitizer_common/.. -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -std=c++11 -Wall -Wextra -Wno-unused-parameter -
Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wcovered-switch-default -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wstring-conversion -fdiagnostics-color -ffunction-s
ections -fdata-sections -Wall -std=c++11 -Wno-unused-parameter -O3 -DNDEBUG     -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -std=c++11 -Wall -Wextra -Wno-unused
-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wcovered-switch-default -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wstring-conversion -fdiagnostics-color 
-ffunction-sections -fdata-sections -Wall -std=c++11 -Wno-unused-parameter -m64 -fPIC -fno-builtin -fno-exceptions -fomit-frame-pointer -funwind-tables -fno-stack-protector -fno-sanitize=safe-stack -fvis
ibility=hidden -fno-lto -O3 -gline-tables-only -Wno-gnu -Wno-variadic-macros -Wno-c99-extensions -Wno-non-virtual-dtor -DFMODE1 -fno-rtti -Wframe-larger-than=570 -Wglobal-constructors -MD -MT projects/co
mpiler-rt/lib/sanitizer_common/CMakeFiles/RTSanitizerCommonNoTermination.x86_64.dir/sanitizer_platform_limits_posix.cc.o -MF projects/compiler-rt/lib/sanitizer_common/CMakeFiles/RTSanitizerCommonNoTermin
ation.x86_64.dir/sanitizer_platform_limits_posix.cc.o.d -o projects/compiler-rt/lib/sanitizer_common/CMakeFiles/RTSanitizerCommonNoTermination.x86_64.dir/sanitizer_platform_limits_posix.cc.o -c /home/adr
ian/Downloads/fuzzers/FuZZan/LLVM/llvm/projects/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cc
/home/adrian/Downloads/fuzzers/FuZZan/LLVM/llvm/projects/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cc:1163:1: error: 'assertion_failed__1163' declared as an array with a negative s
ize
CHECK_SIZE_AND_OFFSET(ipc_perm, mode);
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/adrian/Downloads/fuzzers/FuZZan/LLVM/llvm/projects/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.h:1513:3: note: expanded from macro 'CHECK_SIZE_AND_OFFSET'
  COMPILER_CHECK(sizeof(((__sanitizer_##CLASS *) NULL)->MEMBER) == \
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/adrian/Downloads/fuzzers/FuZZan/LLVM/llvm/projects/compiler-rt/lib/sanitizer_common/sanitizer_internal_defs.h:336:30: note: expanded from macro 'COMPILER_CHECK'
#define COMPILER_CHECK(pred) IMPL_COMPILER_ASSERT(pred, __LINE__)
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/adrian/Downloads/fuzzers/FuZZan/LLVM/llvm/projects/compiler-rt/lib/sanitizer_common/sanitizer_internal_defs.h:342:57: note: expanded from macro 'IMPL_COMPILER_ASSERT'
    typedef char IMPL_PASTE(assertion_failed_##_, line)[2*(int)(pred)-1]
                                                        ^~~~~~~~~~~~~~~
1 error generated.

@gannimo  @jys0710  can I get a bit of help here please? Thank you!

**Comment by jys0710** [comment]
Thanks for your report. I will address this issue by this week.

**Comment by adriantdlg** [comment]
@jys0710  what clang versions have you used to compile the project? thanks.

**Comment by jys0710** [comment]
> @jys0710 what clang versions have you used to compile the project? thanks.

I used LLVM 7.0. Please refer to this file: https://github.com/HexHive/FuZZan/blob/master/LLVM/build.sh

**Comment by adriantdlg** [comment]
@jys0710 

for sure, I used the same script, ran it on Ununtu 20 and got the error I posted above.

**Comment by jys0710** [comment]
Did you try to install FuZZan using Ubuntu 19.10?

**Comment by adriantdlg** [comment]
I can't really downgrade from 20 to 19, thank you.

**Comment by adriantdlg** [comment]
@jys0710 

I tried to set this up in a container running Ubuntu 18 (there's no Ubuntu 19 docker image!) and get this error now:

cc -o libshrink-preload.so libpreload.o shrink.o -shared -Wl,-rpath,xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx -O2 -ldl
ar rcs libshrink-static.a libstatic.o shrink.o
core
performance
/app/FuZZan-master/etc/libshrink
1
kernel.randomize_va_space = 0
/app/FuZZan-master/etc/libshrink
./build.sh: line 17: ../../LLVM/build/bin/clang++: No such file or directory

Looks like clang and clang++ didn't get compiled.

**Comment by jys0710** [comment]
If you don't mind, could you please test FuZZan with VM player and ubuntu 19.10 image first?
- http://old-releases.ubuntu.com/releases/19.10/
FuZZan's porting works (from 19.20 to 20.x) will take time.

**Comment by adrian-tiron** [comment]
@jys0710 ,

I finally found some time for this and trying to set it up on a ubuntu19 as you suggested.
Looks like it gets stuck at the last stage with this error:

cp: cannot stat '../../../Makefile_SRC': No such file or directory
make: *** No rule to make target 'clean'.  Stop.
make: *** No targets specified and no makefile found.  Stop.
make: *** No rule to make target 'install'.  Stop.
/home/adrian/Downloads/FuZZan-master/sample/applications/build_min-16G/tcpdump
cp: cannot stat 'build/sbin/tcpdump': No such file or directory
prelink-tcpdump + 16
Traceback (most recent call last):
  File "/home/adrian/Downloads/FuZZan-master/etc/libshrink/prelink_binary.py", line 10, in <module>
    from elftools.elf.elffile import ELFFile
ImportError: No module named elftools.elf.elffile
/home/adrian/Downloads/FuZZan-master/sample/applications

However, I have those python modules, this is how I tested it:

python3
Python 3.7.5 (default, Apr 19 2020, 20:18:17) 
[GCC 9.2.1 20191008] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import elftools.elf.elffile
>>> 
adrian@ubuntu:~/Downloads/FuZZan-master/sample/applications$ python2 
Python 2.7.17 (default, Nov  7 2019, 10:07:09) 
[GCC 9.2.1 20191008] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import elftools.elf.elffile
>>> 

It imports it just fine, no error.

**Comment by adrian-rt** [comment]
@jys0710 ,

I finally found some time for this and trying to set it up on a ubuntu19 as you suggested.
Looks like it gets stuck at the last stage with this error:

cp: cannot stat '../../../Makefile_SRC': No such file or directory
make: *** No rule to make target 'clean'. Stop.
make: *** No targets specified and no makefile found. Stop.
make: *** No rule to make target 'install'. Stop.
/home/adrian/Downloads/FuZZan-master/sample/applications/build_min-16G/tcpdump
cp: cannot stat 'build/sbin/tcpdump': No such file or directory
prelink-tcpdump + 16
Traceback (most recent call last):
File "/home/adrian/Downloads/FuZZan-master/etc/libshrink/prelink_binary.py", line 10, in
from elftools.elf.elffile import ELFFile
ImportError: No module named elftools.elf.elffile
/home/adrian/Downloads/FuZZan-master/sample/applications

However, I have those python modules, this is how I tested it:

python3
Python 3.7.5 (default, Apr 19 2020, 20:18:17)
[GCC 9.2.1 20191008] on linux
Type "help", "copyright", "credits" or "license" for more information.

import elftools.elf.elffile

adrian@ubuntu:~/Downloads/FuZZan-master/sample/applications$ python2
Python 2.7.17 (default, Nov 7 2019, 10:07:09)
[GCC 9.2.1 20191008] on linux2
Type "help", "copyright", "credits" or "license" for more information.

import elftools.elf.elffile

It imports it just fine, no error.
